src/Controller/ArticlesController.php

```diff
@@ -616,14 +616,24 @@ public function figuresAction(Request $request, string $id, int $version = null)
                     'figures' => $all['figures'],
                     'videos' => $all['videos'],
                     'tables' => $all['tables'],
+                    'data availability' => $all['dataAvailability'],
                     'data sets' => $all['generatedDataSets']->append(...$all['usedDataSets']),
                     'additional files' => $all['additionalFiles'],
                 ], Callback::method('notEmpty'));
             })
             ->then(Callback::mustNotBeEmpty(new NotFoundHttpException('Article version does not contain any figures or data')))
             ->then(function (array $all) {
+                if (1 === count($all) && isset($all['data availability'])) {
+                    return new ViewModel\MessageBar('There are no figures or additional files');
+                }
+
+                unset($all['data availability']);
```

Comment:
Not ideal. Could use a `catch` instead?

Reply:
Or can maybe move the `throw` to the `body`? If that's empty, then the page definitely doesn't exist!